globus_automate_client/flows_client.py

```diff
@@ -700,7 +717,7 @@ def new_client(
         client_id: str,
         authorizer_callback: AuthorizerCallbackType,
         authorizer: AllowedAuthorizersType,
-        base_url: str = PROD_FLOWS_BASE_URL,
+        base_url: Optional[str] = None,
```

Comment:
If we do the `get` above we can deal with a plain `str` type default and not worry about users passing in None, empty strings, or worse...

Reply:
I want a passed in value to override anything based on the `GLOBUS_SDK_ENVIRONMENT`. So, I went to the optional str so that nothing passed will known that using that env var is the next level, and then default from that is our usual prod path.

Does that make sense now that I've explicitly listed the precedence (and does that precedence make sense)?

Reply:
Ah that makes sense. I hadn't thought of that and I think that actually falls in line with the main SDK's behavior. That's a good detail to catch!